node-graph/graph-craft/src/document.rs

```diff
@@ -255,9 +272,13 @@ impl NodeInput {
 
 #[derive(Clone, Debug, PartialEq, Hash, DynAny)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+/// Represents the implementation of a node, which can be a nested [`NodeNetwork`], a proto [`NodeIdentifier`] or extract
 pub enum DocumentNodeImplementation {
+	/// A nested [`NodeNetwork`] that is flattened by the [`NodeNetwork::flatten`] function.
 	Network(NodeNetwork),
+	/// A protonode identifier which can be found in `node_registry.rs`.
 	Unresolved(NodeIdentifier),
+	/// TODO: what?
```

Comment:
This is used for macro functionality and allows you to get the input node as a `DocumentNode` supplied to your node